src/services/data_cache/cache.h

```diff
@@ -11,11 +14,34 @@
  * Used for all getting/setting raw data points
  */
 class Cache {
+private:
+    std::unordered_map<std::string, boost::any> cache;
+    std::vector<std::string> keys;
+
 public:
-    virtual ~Cache() = default;
-    virtual float getFloat(const std::string& key) = 0;
-    virtual float setFloat(const std::string& key, float value) = 0;
-    virtual std::vector<std::string> getKeys() = 0;
+    Cache();
+    ~Cache();
+
+    template <class T>
+    void set(const std::string& key, T value) {
+        if (cache.find(key) == cache.end()) {
+            keys.push_back(key);
+        }
+        cache[key] = value;
+    }
+
+    template <class T>
```

Comment:
I like this API, much cleaner than a bunch of separate functions.

we will need to think carefully about how to cast and save values into the database when we allow any type. Easiest way might be to limit the types to a set list that we know how to serialize/deserialize. Again, something we'll cover when we handle writing data cache values to the database.